src/XRP/shared/rippled-web-socket-schema.ts

```diff
@@ -18,24 +18,29 @@ enum WebSocketReadyState {
  */
 enum RippledMethod {
   subscribe = 'subscribe',
+  accountLines = 'account_lines',
+  gatewayBalances = 'gateway_balances',
 }
 
 /**
- * The standard format for a request to the Web Socket API exposed by a rippled node.
+ * The standard format for a request to the WebSocket API exposed by a rippled node.
  * @see https://xrpl.org/request-formatting.html
  */
-type WebSocketRequestOptions =
+type WebSocketRequest =
   | SubscribeRequest
   | AccountLinesRequest
   | GatewayBalancesRequest
 
+interface BaseRequest {
+  id: number | string
+  command: RippledMethod
+}
+
 /**
  * The standard format for a `subscribe` request to the Web Socket API.
  * @see https://xrpl.org/subscribe.html
  */
-interface SubscribeRequest {
-  id: number | string
-  command: RippledMethod
+interface SubscribeRequest extends BaseRequest {
```

Comment:
can you do

```
command: RippledMethod.subscribe
```
? 

if you do that and add `command: RippledMethod.accountLines` to AccountLinesRequest, ts will be able to tell the difference between `BaseRequest` objects if you switch on `BaseRequest.command`, which is pretty dope

Reply:
Typescript isn't happy when I try to do that and use the objects - `type RippledMethod is not assignable to type RippledMethod.subscribe`